signer/signer.go

```diff
@@ -36,24 +36,34 @@ import (
 // signingAgent is the unprotected header field used by signature.
 const signingAgent = "Notation/1.0.0"
 
-// genericSigner implements notation.Signer and embeds signature.Signer
-type genericSigner struct {
+// GenericSigner implements notation.Signer and embeds signature.Signer
+type GenericSigner struct {
 	signature.Signer
 }
 
 // New returns a builtinSigner given key and cert chain
 func New(key crypto.PrivateKey, certChain []*x509.Certificate) (notation.Signer, error) {
+	return NewGenericSigner(key, certChain)
+}
+
+// NewGenericSigner returns a builtinSigner given key and cert chain
+func NewGenericSigner(key crypto.PrivateKey, certChain []*x509.Certificate) (*GenericSigner, error) {
```

Comment:
Do we need two interfaces for creating GenericSigner from key/cert? Wouldn't that create confusion?

Reply:
Yes this is done for backward comparability, we need to return `GenericSigner` and we cannot update existing function.

Reply:
Should we mark `New` as deprecated so that we encourage users to use NewGenericSigner method?

Reply:
Yep, I can do that in new PR